### PR TITLE
[ie/afreecatv] Adapt extractors to sooplive.co.kr domain

### LINKED ISSUE #11253: [afreecaTV]support for soop
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I *intentionally* remove or skip any mandatory\* field

### Checklist

- [X] I'm reporting a new site support request
- [X] I've verified that I have **updated yt-dlp to nightly or master** ([update instructions](https://github.com/yt-dlp/yt-dlp#update-channels))
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that none of provided URLs [violate any copyrights](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy) or contain any [DRM](https://en.wikipedia.org/wiki/Digital_rights_management) to the best of my knowledge
- [X] I've searched [known issues](https://github.com/yt-dlp/yt-dlp/issues/3766) and the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [ ] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and am willing to share it if required

### Region

South Korea

### Example URLs

https://www.sooplive.co.kr/
- Catch(Shorts like service): https://vod.sooplive.co.kr/player/137220647/catch
- Live: https://play.sooplive.co.kr/vf3366/277793184

### Provide a description that is worded well enough to be understood

Afreeca TV will change its name and domain to Soop starting 10/15/2024 12:00 (KST).
Now, we need to remove the existing Afreeca TV implementation and change to the new implementation for Soop.

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [ ] If using API, add `'verbose': True` to `YoutubeDL` params instead
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
[debug] Command-line config: ['-vU', 'https://play.sooplive.co.kr/vf3366/277793184']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2024.10.07 from yt-dlp/yt-dlp [1a176d874] (source)
[debug] Git HEAD: 1de886665
[debug] Python 3.12.4 (CPython x86_64 64bit) - Linux-6.11.0-8-generic-x86_64-with-glibc2.40 (OpenSSL 3.0.14 4 Jun 2024, glibc 2.40)
[debug] exe versions: ffmpeg 7.0.2 (setts), ffprobe 7.0.2
[debug] Optional libraries: Cryptodome-3.20.0, brotli-1.0.9, certifi-2024.06.02, mutagen-1.47.0, requests-2.32.2, secretstorage-3.3.1, sqlite3-3.45.3, urllib3-2.2.2, websockets-13.0
[debug] Proxy map: {}
[debug] Request Handlers: urllib, requests, websockets
[debug] Loaded 1838 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Latest version: stable@2024.10.07 from yt-dlp/yt-dlp
yt-dlp is up to date (stable@2024.10.07 from yt-dlp/yt-dlp)
[generic] Extracting URL: https://play.sooplive.co.kr/vf3366/277793184
[generic] 277793184: Downloading webpage
WARNING: [generic] Falling back on generic information extractor
[generic] 277793184: Extracting information
[debug] Looking for embeds
[debug] Identified a twitter:player iframe
[generic] Extracting URL: https://play.sooplive.co.kr/vf3366/embed
[generic] embed: Downloading webpage
WARNING: [generic] Falling back on generic information extractor
[generic] embed: Extracting information
[debug] Looking for embeds
ERROR: Unsupported URL: https://play.sooplive.co.kr/vf3366/embed
Traceback (most recent call last):
  File "/data/projects/yt-dlp/yt_dlp/YoutubeDL.py", line 1626, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/data/projects/yt-dlp/yt_dlp/YoutubeDL.py", line 1761, in __extract_info
    ie_result = ie.extract(url)
                ^^^^^^^^^^^^^^^
  File "/data/projects/yt-dlp/yt_dlp/extractor/common.py", line 741, in extract
    ie_result = self._real_extract(url)
                ^^^^^^^^^^^^^^^^^^^^^^^
  File "/data/projects/yt-dlp/yt_dlp/extractor/generic.py", line 2526, in _real_extract
    raise UnsupportedError(url)
yt_dlp.utils.UnsupportedError: Unsupported URL: https://play.sooplive.co.kr/vf3366/embed
```

### PR DESCRIPTION
**Pull request by 63427083**
**IMPORTANT**: PRs without the template will be CLOSED

### Description of your *pull request* and other information

Removed AfreecaTV.
Added Soop.
No "Sooplive" to keep consistency with streamlink.

Fixes #11253

**Preword:**
I had some errors running 'hatch', predominately "EOFError: marshal data too short" for reasons i have yet to figure out.
The code itself works as intended though, most of it is just copy-paste anyway.

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check all of the following options that apply:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)

### What is the purpose of your *pull request*?
- [x] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))

</details>

### CODE REVIEW

**Comment by bashonly** [comment]
are the extractors basically the same as AfreecaTV but with different domains? are the same VOD IDs that were available on AfreecaTV now available on Soop?

**Comment by 63427083** [comment]
> are the extractors basically the same as AfreecaTV but with different domains? are the same VOD IDs that were available on AfreecaTV now available on Soop?

Yes.

**Comment by bashonly** [comment]
do afreecatv URLs now redirect to sooplive?

**Comment by jouven** [comment]
To clear any possible confusion afreecatv.com redirects to sooplive.co.kr. sooplive.com is something else (international version¿), anyway sooplive.com and sooplive.co.kr are different..

**Comment by bashonly** [comment]
The way this should be handled is that we keep the `afreecatv.py` module and we keep the `AfreecaTV` extractors; and only update the `IE_NAME`, `IE_DESC`, `_VALID_URL`s and relevant code

**Comment by 63427083** [comment]
> do afreecatv URLs now redirect to sooplive?

Yes and no.
Most do, basically all the with basedomain sooplive. 
Some kept some weird "afreeca" shortcuts.
Yes, i tested those regarding logins, as i asked myself the same question.

 

> To clear any possible confusion afreecatv.com redirects to sooplive.co.kr. sooplive.com is something else (international version¿), anyway sooplive.com and sooplive.co.kr are different..

This is for sooplive.co.kr. _the_ afreecatv successor. 



> The way this should be handled is that we keep the `afreecatv.py` module and we keep the `AfreecaTV` extractors; and only update the `IE_NAME`, `IE_DESC`, `_VALID_URL`s and relevant code. Do not duplicate / add new `Soop` extractors
> 
> Ideally the `_VALID_URL`s would match both domains for now
> 
> We may need to do a cookie scope fix workaround like we did for Twitter=>X

I'd have to look into that.

**Comment by bashonly** [comment]
> We may need to do a cookie scope fix workaround like we did for Twitter=>X

Don't worry about this for now, I can add a more specific suggestion for it after my other requested changes have been addressed

File: yt_dlp/extractor/afreecatv.py [hunk 1]
```diff
@@ -19,7 +19,7 @@
 
 
 class AfreecaTVBaseIE(InfoExtractor):
-    _NETRC_MACHINE = 'afreecatv'
+    _NETRC_MACHINE = 'soop'
```
Comment (bashonly) [inline yt_dlp/extractor/afreecatv.py:22]: not sure about changing this

File: yt_dlp/extractor/afreecatv.py [hunk 2]
```diff
@@ -61,46 +61,46 @@ def _perform_login(self, username, password):
...
+    IE_DESC = 'sooplive.co.kr'
     _VALID_URL = r'''(?x)
                     https?://
                         (?:
-                            (?:(?:live|afbbs|www)\.)?afreeca(?:tv)?\.com(?::\d+)?
+                            (?:(?:live|afbbs|www)\.)?sooplive\.co\.kr(?::\d+)?
                             (?:
                                 /app/(?:index|read_ucc_bbs)\.cgi|
                                 /player/[Pp]layer\.(?:swf|html)
                             )\?.*?\bnTitleNo=|
```
Comment (bashonly) [inline yt_dlp/extractor/afreecatv.py:80]: do links like these even exist anymore? it seems like all VODs are on the `vod` subdomain now, no?

if we want to keep this part of the regex pattern for legacy reasons or something, we should turn back the clock on the domain name here too
```suggestion
                            (?:(?:live|afbbs|www)\.)?afreecatv.\com(?::\d+)?
                            (?:
                                /app/(?:index|read_ucc_bbs)\.cgi|
                                /player/[Pp]layer\.(?:swf|html)
                            )\?.*?\bnTitleNo=|
```

File: yt_dlp/extractor/afreecatv.py [hunk 3]
```diff
@@ -61,46 +61,46 @@ def _perform_login(self, username, password):
...
-                            vod\.afreecatv\.com/(PLAYER/STATION|player)/
+                            vod\.sooplive\.co\.kr/(PLAYER/STATION|player)/
```
Comment (bashonly) [inline yt_dlp/extractor/afreecatv.py:81]:
```suggestion
                            vod\.(?:sooplive\.co\.kr|afreecatv\.com)/(?:PLAYER/STATION|player)/
```

File: yt_dlp/extractor/afreecatv.py [hunk 4]
```diff
@@ -61,46 +61,46 @@ def _perform_login(self, username, password):
...
                         )
                         (?P<id>\d+)/?(?:$|[?#&])
                     '''
     _TESTS = [{
-        'url': 'http://live.afreecatv.com:8079/app/index.cgi?szType=read_ucc_bbs&szBjId=dailyapril&nStationNo=16711924&nBbsNo=18605867&nTitleNo=36164052&szSkin=',
+        'url': 'http://live.sooplive.co.kr:8079/app/index.cgi?szType=read_ucc_bbs&szBjId=dailyapril&nStationNo=16711924&nBbsNo=18605867&nTitleNo=36164052&szSkin=',
```
Comment (bashonly) [inline yt_dlp/extractor/afreecatv.py:86]: Revert; changing dead URLs to non-existent URLs doesn't make sense. If this type of URL doesn't exist anymore then we should just remove these tests entirely

```suggestion
        'url': 'http://live.afreecatv.com:8079/app/index.cgi?szType=read_ucc_bbs&szBjId=dailyapril&nStationNo=16711924&nBbsNo=18605867&nTitleNo=36164052&szSkin=',
```

File: yt_dlp/extractor/afreecatv.py [hunk 5]
```diff
@@ -61,46 +61,46 @@ def _perform_login(self, username, password):
...
         'md5': 'f72c89fe7ecc14c1b5ce506c4996046e',
         'info_dict': {
             'id': '36164052',
             'ext': 'mp4',
             'title': '데일리 에이프릴 요정들의 시상식!',
-            'thumbnail': 're:^https?://(?:video|st)img.afreecatv.com/.*$',
+            'thumbnail': 're:^https?://(?:video|st)img.sooplive.co.kr/.*$',
```
Comment (bashonly) [inline yt_dlp/extractor/afreecatv.py:92]: should not be changing dead tests
```suggestion
            'thumbnail': 're:^https?://(?:video|st)img.afreecatv.com/.*$',
```

File: yt_dlp/extractor/afreecatv.py [hunk 6]
```diff
@@ -61,46 +61,46 @@ def _perform_login(self, username, password):
...
             'uploader': 'dailyapril',
             'uploader_id': 'dailyapril',
             'upload_date': '20160503',
         },
         'skip': 'Video is gone',
     }, {
-        'url': 'http://afbbs.afreecatv.com:8080/app/read_ucc_bbs.cgi?nStationNo=16711924&nTitleNo=36153164&szBjId=dailyapril&nBbsNo=18605867',
+        'url': 'http://afbbs.sooplive.co.kr:8080/app/read_ucc_bbs.cgi?nStationNo=16711924&nTitleNo=36153164&szBjId=dailyapril&nBbsNo=18605867',
```
Comment (bashonly) [inline yt_dlp/extractor/afreecatv.py:99]: same as above
```suggestion
        'url': 'http://afbbs.afreecatv.com:8080/app/read_ucc_bbs.cgi?nStationNo=16711924&nTitleNo=36153164&szBjId=dailyapril&nBbsNo=18605867',
```

File: yt_dlp/extractor/afreecatv.py [hunk 7]
```diff
@@ -61,46 +61,46 @@ def _perform_login(self, username, password):
...
         'info_dict': {
             'id': '36153164',
             'title': "BJ유트루와 함께하는 '팅커벨 메이크업!'",
-            'thumbnail': 're:^https?://(?:video|st)img.afreecatv.com/.*$',
+            'thumbnail': 're:^https?://(?:video|st)img.sooplive.co.kr/.*$',
```
Comment (bashonly) [inline yt_dlp/extractor/afreecatv.py:103]:
```suggestion
            'thumbnail': 're:^https?://(?:video|st)img.afreecatv.com/.*$',
```

File: yt_dlp/extractor/afreecatv.py [hunk 8]
```diff
@@ -158,17 +158,17 @@ class AfreecaTVIE(AfreecaTVBaseIE):
         },
         'skip': 'The VOD does not exist',
     }, {
-        'url': 'http://www.afreecatv.com/player/Player.swf?szType=szBjId=djleegoon&nStationNo=11273158&nBbsNo=13161095&nTitleNo=36327652',
+        'url': 'http://www.sooplive.co.kr/player/Player.swf?szType=szBjId=djleegoon&nStationNo=11273158&nBbsNo=13161095&nTitleNo=36327652',
```
Comment (bashonly) [inline yt_dlp/extractor/afreecatv.py:161]:
```suggestion
        'url': 'http://www.afreecatv.com/player/Player.swf?szType=szBjId=djleegoon&nStationNo=11273158&nBbsNo=13161095&nTitleNo=36327652',
```

File: yt_dlp/extractor/afreecatv.py [hunk 9]
```diff
@@ -262,11 +262,11 @@ def _real_extract(self, url):
 
 
 class AfreecaTVCatchStoryIE(AfreecaTVBaseIE):
-    IE_NAME = 'afreecatv:catchstory'
-    IE_DESC = 'afreecatv.com catch story'
-    _VALID_URL = r'https?://vod\.afreecatv\.com/player/(?P<id>\d+)/catchstory'
+    IE_NAME = 'soop:catchstory'
+    IE_DESC = 'sooplive.co.kr catch story'
+    _VALID_URL = r'https?://vod\.sooplive\.co\.kr/player/(?P<id>\d+)/catchstory'
```
Comment (bashonly) [inline yt_dlp/extractor/afreecatv.py:267]:
```suggestion
    _VALID_URL = r'https?://vod\.(?:sooplive\.co\.kr|afreecatv\.com)/player/(?P<id>\d+)/catchstory'
```

File: yt_dlp/extractor/afreecatv.py [hunk 10]
```diff
@@ -299,11 +299,11 @@ def _entries(data):
 
 
 class AfreecaTVLiveIE(AfreecaTVBaseIE):
-    IE_NAME = 'afreecatv:live'
-    IE_DESC = 'afreecatv.com livestreams'
-    _VALID_URL = r'https?://play\.afreeca(?:tv)?\.com/(?P<id>[^/]+)(?:/(?P<bno>\d+))?'
+    IE_NAME = 'soop:live'
+    IE_DESC = 'sooplive.co.kr livestreams'
+    _VALID_URL = r'https?://play\.sooplive\.co\.kr/(?P<id>[^/]+)(?:/(?P<bno>\d+))?'
```
Comment (bashonly) [inline yt_dlp/extractor/afreecatv.py:304]:
```suggestion
    _VALID_URL = r'https?://play\.(?:sooplive\.co\.kr|afreecatv\.com)/(?P<id>[^/?#]+)(?:/(?P<bno>\d+))?'
```

File: yt_dlp/extractor/afreecatv.py [hunk 11]
```diff
@@ -315,30 +315,30 @@ class AfreecaTVLiveIE(AfreecaTVBaseIE):
...
+    _LIVE_API_URL = 'https://live.sooplive.co.kr/afreeca/player_live_api.php'
     _WORKING_CDNS = [
-        'gcp_cdn',  # live-global-cdn-v02.afreecatv.com
-        'gs_cdn_pc_app',  # pc-app.stream.afreecatv.com
-        'gs_cdn_mobile_web',  # mobile-web.stream.afreecatv.com
-        'gs_cdn_pc_web',  # pc-web.stream.afreecatv.com
+        'gcp_cdn',  # live-global-cdn-v02.sooplive.co.kr
+        'gs_cdn_pc_app',  # pc-app.stream.sooplive.co.kr
+        'gs_cdn_mobile_web',  # mobile-web.stream.sooplive.co.kr
+        'gs_cdn_pc_web',  # pc-web.stream.sooplive.co.kr
```
Comment (bashonly) [inline yt_dlp/extractor/afreecatv.py:330]: are all these CDN domains confirmed?

File: yt_dlp/extractor/afreecatv.py [hunk 12]
```diff
@@ -315,30 +315,30 @@ class AfreecaTVLiveIE(AfreecaTVBaseIE):
...
     _BAD_CDNS = [
         'gs_cdn',  # chromecast.afreeca.gscdn.com (cannot resolve)
-        'gs_cdn_chromecast',  # chromecast.stream.afreecatv.com (HTTP Error 400)
-        'azure_cdn',  # live-global-cdn-v01.afreecatv.com (cannot resolve)
-        'aws_cf',  # live-global-cdn-v03.afreecatv.com (cannot resolve)
-        'kt_cdn',  # kt.stream.afreecatv.com (HTTP Error 400)
+        'gs_cdn_chromecast',  # chromecast.stream.sooplive.co.kr (HTTP Error 400)
+        'azure_cdn',  # live-global-cdn-v01.sooplive.co.kr (cannot resolve)
+        'aws_cf',  # live-global-cdn-v03.sooplive.co.kr (cannot resolve)
+        'kt_cdn',  # kt.stream.sooplive.co.kr (HTTP Error 400)
```
Comment (bashonly) [inline yt_dlp/extractor/afreecatv.py:337]: same question here

File: yt_dlp/extractor/afreecatv.py [hunk 13]
```diff
@@ -420,34 +420,34 @@ def _real_extract(self, url):
 
 
 class AfreecaTVUserIE(InfoExtractor):
-    IE_NAME = 'afreecatv:user'
-    _VALID_URL = r'https?://bj\.afreeca(?:tv)?\.com/(?P<id>[^/]+)/vods/?(?P<slug_type>[^/]+)?'
+    IE_NAME = 'soop:user'
+    _VALID_URL = r'https?://ch\.sooplive\.co\.kr/(?P<id>[^/]+)/vods/?(?P<slug_type>[^/]+)?'
```
Comment (bashonly) [inline yt_dlp/extractor/afreecatv.py:424]:
```suggestion
    _VALID_URL = r'https?://ch\.(?:sooplive\.co\.kr|afreecatv\.com)/(?P<id>[^/?#]+)/vods/?(?P<slug_type>[^/?#]+)?'
```

**Review by bashonly — CHANGES_REQUESTED** [review]
I partially reverted your initial commit to comply with my request. It should be pretty easy to finish this one up now. I do have some questions / further suggestions though

**Comment by bashonly** [comment]
cc @hui1601 would you like to take a look at this?

**Comment by hui1601** [comment]
> No “Sooplive” to keep consistency with streamlink.

In my opinion, there is already a PR (#10136) to add `SoopGlobal` extractor, so I think it should be changed to `SoopKorea` or `Soop` or something like that, for future code management.
Is there any reason why we need to match the name of the streamlink?

File: yt_dlp/extractor/afreecatv.py [hunk 14]
```diff
@@ -61,24 +61,24 @@ def _perform_login(self, username, password):
...
 class AfreecaTVIE(AfreecaTVBaseIE):
-    IE_NAME = 'afreecatv'
-    IE_DESC = 'afreecatv.com'
+    IE_NAME = 'soop'
+    IE_DESC = 'sooplive.co.kr'
     _VALID_URL = r'''(?x)
                     https?://
                         (?:
-                            (?:(?:live|afbbs|www)\.)?afreeca(?:tv)?\.com(?::\d+)?
+                            (?:(?:live|afbbs|www)\.)?afreecatv.\com(?::\d+)?
```
Comment (bashonly) [inline yt_dlp/extractor/afreecatv.py:76]:
```suggestion
                            (?:(?:live|afbbs|www)\.)?afreecatv\.com(?::\d+)?
```

File: yt_dlp/extractor/afreecatv.py (hunk 1 above)
Comment (bashonly) [reply yt_dlp/extractor/afreecatv.py:22]:
```suggestion
    _NETRC_MACHINE = 'afreecatv'
```
we can change in the future, for now let's keep it the same to not break users' netrc files

File: yt_dlp/extractor/afreecatv.py [hunk 15]
```diff
@@ -61,76 +61,40 @@ def _perform_login(self, username, password):
...
     }, {
         # non standard key
-        'url': 'http://vod.afreecatv.com/PLAYER/STATION/20515605',
+        'url': 'http://vod.sooplive.co.kr/PLAYER/STATION/20515605',
         'info_dict': {
             'id': '20170411_BE689A0E_190960999_1_2_h',
             'ext': 'mp4',
             'title': '혼자사는여자집',
-            'thumbnail': 're:^https?://(?:video|st)img.afreecatv.com/.*$',
+            'thumbnail': 're:https?://(?:video|st)img\.sooplive\.co\.kr/.+',
```
Comment (bashonly) [inline yt_dlp/extractor/afreecatv.py:97]:
```suggestion
            'thumbnail': r're:https?://(?:video|st)img\.sooplive\.co\.kr/.+',
```

File: yt_dlp/extractor/afreecatv.py [hunk 16]
```diff
@@ -142,12 +106,12 @@ class AfreecaTVIE(AfreecaTVBaseIE):
...
     }, {
         # adult content
-        'url': 'https://vod.afreecatv.com/player/97267690',
+        'url': 'https://vod.sooplive.co.kr/player/97267690',
         'info_dict': {
             'id': '20180327_27901457_202289533_1',
             'ext': 'mp4',
             'title': '[생]빨개요♥ (part 1)',
-            'thumbnail': 're:^https?://(?:video|st)img.afreecatv.com/.*$',
+            'thumbnail': 're:https?://(?:video|st)img\.sooplive\.co\.kr/.+',
```
Comment (bashonly) [inline yt_dlp/extractor/afreecatv.py:114]:
```suggestion
            'thumbnail': r're:https?://(?:video|st)img\.sooplive\.co\.kr/.+',
```

**Review by bashonly — APPROVED** [review]
I think this looks good now. As for the `IE_NAME`, we can adjust that as needed when finishing up and merging the soopglobal PR. It's not a big deal to change the `IE_NAME`, not like how changing the extractor class name is

**Review by seproDev — APPROVED** [review]

**Comment by bashonly** [comment]
Also, I'm not implementing a cookies migration domain-switch workaround like we did for Twitter because I have faith in this site having a higher competency level than Twitter. If we get reports of problems, then we can add the domain switch hack

**Comment by NormanMises** [comment]
请问有支持 sooplive.com 的计划吗？我想下载上面的视频，比如 https:www.sooplive.com/video/121620

Is there a plan to support sooplive.com? I want to download videos from there, for example https:www.sooplive.com/video/121620

**Comment by hui1601** [comment]
> 请问有支持 sooplive.com 的计划吗？我想下载上面的视频，比如 https:www.sooplive.com/video/121620
> 
> Is there a plan to support sooplive.com? I want to download videos from there, for example https:www.sooplive.com/video/121620
> @NormanMises 

There is already an issue about this.
See #10066.

**Comment by NormanMises** [comment]
> > 请问有支持 sooplive.com 的计划吗？我想下载上面的视频，比如 https:www.sooplive.com/video/121620
> > Is there a plan to support sooplive.com? I want to download videos from there, for example https:www.sooplive.com/video/121620
> > @NormanMises
> 
> There is already an issue about this. See #10066.

thanks！！！ Truly Need it！！！

**Pull request merged**